test/gtest/error_handling.cpp

```diff
@@ -258,12 +284,24 @@ bool TestErrorHandling::isFailure(size_t iter) {
     switch (test_type) {
     case TestType::BASIC_XFER:            return false;
     case TestType::LOAD_REMOTE_THEN_FAIL: return iter == 0;
-    case TestType::XFER_THEN_FAIL:        return iter == 1;
+    case TestType::XFER_THEN_FAIL:
+    case TestType::XFER_FAIL_RESTORE:
+        return iter == 1;
     }
 }
 
-template<TestErrorHandling::TestType test_type> size_t TestErrorHandling::numIter() {
-    return (test_type == TestType::XFER_THEN_FAIL) ? 2 : 1;
+template<TestErrorHandling::TestType test_type>
+size_t
+TestErrorHandling::numIter() {
```

Comment:
Why do we need here a template? Can we just pass `test_type` as a parameter?

Reply:
to calculate `numIter()` in compile time

Reply:
I see. I would assume that the compiler will do the same, since a constant value is passed in the code calling the function.